libcudacxx/include/cuda/__memcpy_async/elect_one.h

```diff
@@ -22,37 +22,36 @@
 #  pragma system_header
 #endif // no system header
 
-#if _CCCL_CUDA_COMPILATION()
-#  if __cccl_ptx_isa >= 800
+#if _CCCL_CUDA_COMPILATION() && __cccl_ptx_isa >= 800
+#  include <cuda/__ptx/instructions/elect_sync.h>
+#endif // _CCCL_CUDA_COMPILATION() && __cccl_ptx_isa >= 800
 
-#    include <cuda/__ptx/instructions/elect_sync.h>
-
-#    include <cuda/std/__cccl/prologue.h>
+#include <cuda/std/__cccl/prologue.h>
 
 _CCCL_BEGIN_NAMESPACE_CUDA_DEVICE
 
-extern "C" _CCCL_DEVICE void __cuda_elect_sync_is_not_supported_before_SM_90__();
-
-//! Elects a single leader thread from a one dimensional thread block
-[[nodiscard]] _CCCL_DEVICE_API _CCCL_FORCEINLINE bool __elect_one() noexcept
+//! Elects a single leader thread from a one dimensional thread block. For SM90+ will use ptx::elect_sync() etc.,
+//! otherwise just selects the thread with ID 0. If the returned value is used as condition for an if statement, the
+//! compiler will emit a uniform data path for the branch.
+[[nodiscard]] _CCCL_DEVICE_API _CCCL_FORCEINLINE bool __block_elect_one() noexcept
 {
   _CCCL_ASSERT(blockDim.y == 1 && blockDim.z == 1, "The block must by one dimensional");
 
-  NV_IF_ELSE_TARGET(
+#if _CCCL_CUDA_COMPILATION() && __cccl_ptx_isa >= 800
+  NV_IF_TARGET(
     NV_PROVIDES_SM_90,
     (const auto tid             = threadIdx.x; //
      const auto warp_id         = tid / 32;
      const auto uniform_warp_id = __shfl_sync(~0, warp_id, 0); // broadcast from lane 0
      return uniform_warp_id == 0 && cuda::ptx::elect_sync(~0); // elect a leader thread among warp 0
-     ),
-    (::cuda::device::__cuda_elect_sync_is_not_supported_before_SM_90__(); _CCCL_UNREACHABLE();));
```

Comment:
Shouldn't the 
```
  return threadIdx.x == 0;
```
go in the else here? 

AFAIK PTX ISA is CTK 12.0. CCCL support 12.0 and up, so there is no need for the `__cccl_ptx_isa` ifdef.